app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -121,16 +125,66 @@ public void initializeRecyclerView(LinearLayoutManager layoutManager, View view)
             @Override
             // if row at the specified position is clicked
             public void onRowClicked(int position) {
-                openViewWindowForItem(position);
+                // check if a row has been long pressed
+                if (_longPressed) {
+                    // store the to and from positions
+                    int fromPosition = _longPressedPosition;
+                    int toPosition = position;
+
+                    // get the correspondong habits
+                    Habit fromHabit = _habitItemAdapter.getItem(fromPosition);
+                    Habit toHabit = _habitItemAdapter.getItem(toPosition);
+
+                    // change habit item background color back to white
+                    recyclerView.findViewHolderForAdapterPosition(_longPressedPosition).itemView.findViewById(R.id.habit_view_constraint).setBackgroundColor(Color.WHITE);
+
+                    // swap the two habits
+                    swapSortIndex(fromHabit, toHabit, fromPosition, toPosition);
+                    _longPressed = false;
+                } else {
+                    openViewWindowForItem(position);
+                }
             }
         })
                 .setSwipeOptionViews(R.id.edit_button, R.id.delete_button)
                 .setSwipeable(R.id.habit_view, R.id.swipe_options,
                         new SwipeListener(this));
+
+        touchListener.setLongClickable(false, new RecyclerTouchListener.OnRowLongClickListener() {
+
+            // this checks for long presses
+            @Override
+            public void onRowLongClicked(int longPosition) {
+                // save position of long pressed habit item
+                _longPressedPosition = longPosition;
+                _longPressed = true;
+                // change background color of long pressed item
+                recyclerView.findViewHolderForAdapterPosition(longPosition).itemView.findViewById(R.id.habit_view_constraint).setBackgroundColor(Color.GRAY);
+                // send toast message to remind user of how to swap habits
+                Toast.makeText(_context, "Please select another Habit to swap with.", Toast.LENGTH_LONG).show();
```

Comment:
Likewise, we should probably define how we handle the case where a row has already been long pressed.

- [ ] Handle case where a row has already been long pressed